test/functional/apps/visualize/_inspector.ts

```diff
@@ -34,6 +37,45 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       await PageObjects.timePicker.setDefaultAbsoluteRange();
     });
 
+    describe('advanced input JSON', () => {
+      it('should match JSON string', async () => {
+        log.debug('Add Max Metric on memory field');
+        await PageObjects.visEditor.clickBucket('Y-axis', 'metrics');
+        await PageObjects.visEditor.selectAggregation('Max', 'metrics');
+        await PageObjects.visEditor.selectField('memory', 'metrics');
+
+        log.debug('Add value to advanced JSON input');
+        await PageObjects.visEditor.toggleAdvancedParams('2');
+        await testSubjects.setValue('codeEditorContainer', '{ "missing": 10 }');
+        await PageObjects.visEditor.clickGo();
+
+        await inspector.open();
+        await inspector.openInspectorRequestsView();
+        const requestTab = await inspector.getOpenRequestDetailRequestButton();
+        await requestTab.click();
+        const requestJSON = await inspector.getCodeEditorValue();
+
+        expect(requestJSON).to.be(
+          '{\n  "aggs": {\n    "2": {\n      "max": {\n        "field": "memory",\n        "missing": 10\n      }\n' +
+            '    }\n  },\n  "size": 0,\n  "fields": [\n    {\n      "field": "@timestamp",\n      "format": "date_time"\n' +
+            '    },\n    {\n      "field": "relatedContent.article:modified_time",\n      "format": "date_time"\n    },\n' +
+            '    {\n      "field": "relatedContent.article:published_time",\n      "format": "date_time"\n    },\n    {\n' +
+            '      "field": "utc_time",\n      "format": "date_time"\n    }\n  ],\n  "script_fields": {},\n  "stored_fields": ' +
+            '[\n    "*"\n  ],\n  "_source": {\n    "excludes": []\n  },\n  "query": {\n    "bool": {\n      "must": [],\n      ' +
+            '"filter": [\n        {\n          "match_all": {}\n        },\n        {\n          "range": {\n            ' +
+            '"@timestamp": {\n              "gte": "2015-09-19T06:31:44.000Z",\n              "lte": "2015-09-23T18:31:44.000Z",\n' +
+            '              "format": "strict_date_optional_time"\n            }\n          }\n        }\n      ],\n      ' +
+            '"should": [],\n      "must_not": []\n    }\n  }\n}'
+        );
+      });
+
+      after(async () => {
```

Comment:
It's unusual for me to see the “after” block without the “before”. I can't say that this is wrong, because I understand that you want to close the inspector view regardless of the result of expect.